src/main/java/com/esri/arcgisruntime/toolkit/skins/FeatureTemplatePickerTilePaneSkin.java

```diff
@@ -0,0 +1,329 @@
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.*;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.TilePane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Screen;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+/**
+ * <p>A skin for displaying a {@link FeatureTemplatePicker} with tile panes. Feature template groups correspond to
+ * tile panes and feature template items correspond to toggle buttons within the tile panes.</p>
+ *
+ * <p> If the picker's orientation is horizontal, tile panes will be aligned left to right. If vertical, tile panes
+ * are aligned top to bottom.</p>
+ *
+ * <p>The symbol size determines the height and width in dp of the symbol swatch. Toggle buttons will have a
+ * preferred size of 2 * symbol size. Template names which overflow this size will be clipped with ellipsis.</p>
+ *
+ * <p>The following css selectors can be used to style the picker:</p>
+ *
+ * <ul>
+ * <li>.feature-template-picker: styles the content node of the scroll pane</li>
+ * <li>.feature-template-group: styles the vbox containing the feature template group label and tile pane</li>
+ * <li>.tile-pane: style the tile pane in a feature template group</li>
+ * <li>.feature-template-item: styles the toggle button</li></p>
+ * </ul>
+ *
+ * <p>A "horizontal" pseudo class is added to the template picker when its orientation is horizontal. The following
+ * snippet shows a selector for applying a style to feature template groups only when the picker has a horizontal
+ * orientation:</p>
+ * <pre>
+ * .feature-template-picker:horizontal .feature-template-group {
+ *
+ * }
+ * </pre>
+ *
+ * @since 100.7.0
+ */
+public final class FeatureTemplatePickerTilePaneSkin extends SkinBase<FeatureTemplatePicker> {
+
+  private static final PseudoClass HORIZONTAL_PSEUDO_CLASS = PseudoClass.getPseudoClass("horizontal");
+
+  private Pane contentPane = new VBox();
+  private final ScrollPane scrollPane = new ScrollPane();
+  private final ToggleGroup toggleGroup = new ToggleGroup();
+  private boolean swatchesNeedUpdate = false;
+
+  public FeatureTemplatePickerTilePaneSkin(FeatureTemplatePicker control) {
+    super(control);
+    // put everything in a scroll pane to handle sizing within the skin
+    getChildren().add(scrollPane);
+
+    updateOrientation();
+
+    // use a shared toggle group for all feature template items to keep track of the current selected item
+    toggleGroup.selectedToggleProperty().addListener((observable, oldValue, newValue) ->
+      control.setSelectedFeatureTemplateItem(newValue == null ? null : (FeatureTemplateItem) newValue.getUserData())
+    );
+    control.selectedFeatureTemplateItemProperty().addListener((observable, oldValue, newValue) -> {
```

Comment:
I wonder if it should be an exception to set the selected template item with an item that isn't actually a member of the toggle group i.e. an item from a different template picker.

Reply:
The SelectionModel javadoc seems to suggest an exception is not thrown if the item cannot be found: https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/SelectionModel.html#select-T-.

Reply:
Done